src/lib/legends/discrete-color-legend.js

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React from 'react';
+
+import DiscreteColorLegendItem from './discrete-color-legend-item';
+
+const propTypes = {
+  items: React.PropTypes.arrayOf(
+    React.PropTypes.shape(DiscreteColorLegendItem.propTypes)
+  ),
+  onItemClick: React.PropTypes.func,
+  height: React.PropTypes.number,
+  width: React.PropTypes.number
+};
+
+function DiscreteColorLegend({items, width, height, onItemClick}) {
+  return (
+    <div className="rv-discrete-color-legend" style={{width, height}}>
+      {items.map((item, i) =>
+        <DiscreteColorLegendItem key={i}
+          {...item}
+          onClick={onItemClick ? () => onItemClick(item, i) : null}
```

Comment:
Yes, this definitely should be handled, but only at the situations when the legend is a part of some chart. Currently legends don't provide this functionality, so they should be wired to the chart manually